core/state_processor.go

```diff
@@ -73,12 +73,15 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
 	if p.config.DAOForkSupport && p.config.DAOForkBlock != nil && p.config.DAOForkBlock.Cmp(block.Number()) == 0 {
 		misc.ApplyDAOHardFork(statedb)
 	}
-	// Handle upgrade build-in system contract code
-	lastBlock := p.bc.GetBlockByHash(block.ParentHash())
-	if lastBlock == nil {
-		return statedb, nil, nil, 0, fmt.Errorf("could not get parent block")
+
+	if !p.config.IsFeynman(block.Number(), block.Time()) {
+		// Handle upgrade build-in system contract code
+		lastBlock := p.bc.GetBlockByHash(block.ParentHash())
+		if lastBlock == nil {
+			return statedb, nil, nil, 0, fmt.Errorf("could not get parent block")
+		}
```

Comment:
we can move line 79 -> line 82 out of the if scope, i.e. check lastBlock before Feynman check

Reply:
Done